src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -7191,19 +7202,20 @@ bool Unit::HandleDummyAuraProc(Unit* victim, uint32 damage, AuraEffect* triggere
                 if (effIndex != 1)
                     return false;
 
-                Unit* pPet = NULL;
-                for (ControlList::const_iterator itr = m_Controlled.begin(); itr != m_Controlled.end(); ++itr) // Find Rune Weapon
-                    if ((*itr)->GetEntry() == 27893)
+                Unit* runeWeapon = nullptr;
+                for (ControlList::const_iterator itr = m_Controlled.begin(); itr != m_Controlled.end() && !runeWeapon; itr++)
+                    if ((*itr)->GetEntry() == 27893 /*Dancing Rune Weapon*/)
                     {
-                        pPet = *itr;
-                        break;
+                        runeWeapon = *itr;
                     }
 
-                if (pPet && pPet->GetVictim() && damage && procSpell)
+                if (runeWeapon && runeWeapon->GetVictim() && damage && procSpell)
                 {
-                    uint32 procDmg = damage / 2;
+                    /*uint32 procDmg = damage / 2;
                     pPet->SendSpellNonMeleeDamageLog(pPet->GetVictim(), procSpell->Id, procDmg, procSpell->GetSchoolMask(), 0, 0, false, 0, false);
-                    pPet->DealDamage(pPet->GetVictim(), procDmg, NULL, SPELL_DIRECT_DAMAGE, procSpell->GetSchoolMask(), procSpell, true);
+                    pPet->DealDamage(pPet->GetVictim(), procDmg, NULL, SPELL_DIRECT_DAMAGE, procSpell->GetSchoolMask(), procSpell, true);*/
+
+                    // ToDo: use spell cast
```

Comment:
Yes, it's a WIP.

I opened it to obtain some feedback while I'm still working on it.